Use more smart pointers in Source/WebKit

cdumez opened the pull request:
#### bd0e96182943cd6e05b1f15f771ca926832e42d3
<pre>
Use more smart pointers in Source/WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=261474">https://bugs.webkit.org/show_bug.cgi?id=261474</a>

Reviewed by Brent Fulgham.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::focusedElementChanged):
(WebKit::WebChromeClient::focusedFrameChanged):
(WebKit::WebChromeClient::createWindow):
(WebKit::WebChromeClient::runBeforeUnloadConfirmPanel):
(WebKit::WebChromeClient::closeWindow):
(WebKit::WebChromeClient::runJavaScriptAlert):
(WebKit::WebChromeClient::runJavaScriptConfirm):
(WebKit::WebChromeClient::runJavaScriptPrompt):
(WebKit::WebChromeClient::invalidateContentsAndRootView):
(WebKit::WebChromeClient::invalidateContentsForSlowScroll):
(WebKit::WebChromeClient::contentsSizeChanged const):
(WebKit::WebChromeClient::print):
(WebKit::WebChromeClient::runOpenPanel):
(WebKit::WebChromeClient::didAssociateFormControls):
(WebKit::WebChromeClient::showPlaybackTargetPicker):
(WebKit::WebChromeClient::hasStorageAccess):
(WebKit::WebChromeClient::requestStorageAccess):
(WebKit::WebChromeClient::shouldAllowDeviceOrientationAndMotionAccess):
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.cpp:
(WebKit::WebEditorClient::textFieldDidBeginEditing):
(WebKit::WebEditorClient::textFieldDidEndEditing):
(WebKit::WebEditorClient::textDidChangeInTextField):
(WebKit::WebEditorClient::textDidChangeInTextArea):
(WebKit::WebEditorClient::doTextFieldCommandFromEvent):
(WebKit::WebEditorClient::textWillBeDeletedInTextField):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction):
* Source/WebKit/WebProcess/WebCoreSupport/WebGeolocationClient.cpp:
(WebKit::WebGeolocationClient::geolocationDestroyed):
(WebKit::WebGeolocationClient::startUpdating):
(WebKit::WebGeolocationClient::stopUpdating):
(WebKit::WebGeolocationClient::setEnableHighAccuracy):
(WebKit::WebGeolocationClient::requestPermission):
(WebKit::WebGeolocationClient::revokeAuthorizationToken):
(WebKit::WebGeolocationClient::cancelPermissionRequest):
* Source/WebKit/WebProcess/WebCoreSupport/WebGeolocationClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didLoadFromRegistrableDomain):
(WebKit::WebLocalFrameLoaderClient::loadedSubresourceDomains const):
(WebKit::WebLocalFrameLoaderClient::detachedFromParent2):
(WebKit::WebLocalFrameLoaderClient::assignIdentifierToInitialRequest):
(WebKit::WebLocalFrameLoaderClient::dispatchWillSendRequest):
(WebKit::WebLocalFrameLoaderClient::shouldUseCredentialStorage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveResponse):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveContentLength):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishDataDetection):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishLoading):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFailLoading):
(WebKit::WebLocalFrameLoaderClient::dispatchDidDispatchOnloadEvents):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveServerRedirectForProvisionalLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidChangeProvisionalURL):
(WebKit::WebLocalFrameLoaderClient::dispatchDidCancelClientRedirect):
(WebKit::WebLocalFrameLoaderClient::dispatchWillPerformClientRedirect):
(WebKit::WebLocalFrameLoaderClient::dispatchDidChangeLocationWithinPage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidChangeMainDocument):
(WebKit::WebLocalFrameLoaderClient::dispatchWillChangeDocument):
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI):
(WebKit::WebLocalFrameLoaderClient::dispatchDidExplicitOpen):
(WebKit::WebLocalFrameLoaderClient::dispatchDidStartProvisionalLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveTitle):
(WebKit::WebLocalFrameLoaderClient::dispatchDidCommitLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFailProvisionalLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFailLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishDocumentLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishLoad):
(WebKit::WebLocalFrameLoaderClient::completePageTransitionIfNeeded):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReachLayoutMilestone):
(WebKit::WebLocalFrameLoaderClient::dispatchDidLayout):
(WebKit::WebLocalFrameLoaderClient::dispatchCreatePage):
(WebKit::WebLocalFrameLoaderClient::dispatchShow):
(WebKit::WebLocalFrameLoaderClient::dispatchDecidePolicyForResponse):
(WebKit::WebLocalFrameLoaderClient::dispatchDecidePolicyForNewWindowAction):
(WebKit::WebLocalFrameLoaderClient::allowsContentJavaScriptFromMostRecentNavigation const):
(WebKit::WebLocalFrameLoaderClient::dispatchWillSendSubmitEvent):
(WebKit::WebLocalFrameLoaderClient::dispatchWillSubmitForm):
(WebKit::WebLocalFrameLoaderClient::willReplaceMultipartContent):
(WebKit::WebLocalFrameLoaderClient::didReplaceMultipartContent):
(WebKit::WebLocalFrameLoaderClient::finishedLoading):
(WebKit::WebLocalFrameLoaderClient::updateGlobalHistory):
(WebKit::WebLocalFrameLoaderClient::updateGlobalHistoryRedirectLinks):
(WebKit::WebLocalFrameLoaderClient::shouldGoToHistoryItem const):
(WebKit::WebLocalFrameLoaderClient::didDisplayInsecureContent):
(WebKit::WebLocalFrameLoaderClient::didRunInsecureContent):
(WebKit::WebLocalFrameLoaderClient::provisionalLoadStarted):
(WebKit::WebLocalFrameLoaderClient::setTitle):
(WebKit::WebLocalFrameLoaderClient::userAgent const):
(WebKit::WebLocalFrameLoaderClient::overrideContentSecurityPolicy const):
(WebKit::WebLocalFrameLoaderClient::transitionToCommittedForNewPage):
(WebKit::WebLocalFrameLoaderClient::createFrame):
(WebKit::WebLocalFrameLoaderClient::objectContentType):
(WebKit::WebLocalFrameLoaderClient::dispatchDidClearWindowObjectInWorld):
(WebKit::WebLocalFrameLoaderClient::dispatchGlobalObjectAvailable):
(WebKit::WebLocalFrameLoaderClient::dispatchServiceWorkerGlobalObjectAvailable):
(WebKit::WebLocalFrameLoaderClient::willInjectUserScript):
(WebKit::WebLocalFrameLoaderClient::dispatchWillDisconnectDOMWindowExtensionFromGlobalObject):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReconnectDOMWindowExtensionToGlobalObject):
(WebKit::WebLocalFrameLoaderClient::dispatchWillDestroyGlobalObjectForDOMWindowExtension):
(WebKit::WebLocalFrameLoaderClient::accessibilityRemoteObject):
(WebKit::WebLocalFrameLoaderClient::setAXIsolatedTreeRoot):
(WebKit::WebLocalFrameLoaderClient::willCacheResponse const):
(WebKit::WebLocalFrameLoaderClient::dataDetectionReferenceDate):
(WebKit::WebLocalFrameLoaderClient::didChangeScrollOffset):
(WebKit::WebLocalFrameLoaderClient::shouldForceUniversalAccessFromLocalURL):
(WebKit::WebLocalFrameLoaderClient::sendH2Ping):
(WebKit::WebLocalFrameLoaderClient::didRestoreScrollPosition):
(WebKit::WebLocalFrameLoaderClient::getLoadDecisionForIcons):
(WebKit::WebLocalFrameLoaderClient::didFinishServiceWorkerPageRegistration):
(WebKit::WebLocalFrameLoaderClient::notifyPageOfAppBoundBehavior):
(WebKit::WebLocalFrameLoaderClient::modelInlinePreviewUUIDs const):
* Source/WebKit/WebProcess/WebCoreSupport/WebValidationMessageClient.cpp:
(WebKit::WebValidationMessageClient::~WebValidationMessageClient):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebRemoteObjectRegistry.cpp:
(WebKit::WebRemoteObjectRegistry::WebRemoteObjectRegistry):
(WebKit::WebRemoteObjectRegistry::close):
(WebKit::WebRemoteObjectRegistry::messageSender):
(WebKit::WebRemoteObjectRegistry::messageDestinationID):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebRemoteObjectRegistry.h:
* Source/WebKit/WebProcess/WebPage/WebDocumentLoader.cpp:
(WebKit::WebDocumentLoader::loaderForWebsitePolicies):

Canonical link: <a href="https://commits.webkit.org/267945@main">https://commits.webkit.org/267945@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/09f7e49d699a80e4397e2a67fa8388e4efc19f52

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18120 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18453 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19020 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19958 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16959 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21750 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18610 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18924 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18339 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18578 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15772 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20836 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15802 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23037 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16821 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16696 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20918 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17266 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-WK2-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16357 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4319 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20718 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17114 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
```diff
@@ -313,7 +313,7 @@ Page* WebChromeClient::createWindow(LocalFrame& frame, const WindowFeatures& win
     };
 
-    WebFrame* webFrame = WebFrame::fromCoreFrame(frame);
+    RefPtr webFrame = WebFrame::fromCoreFrame(frame);
```
Should `WebFrame::fromCoreFrame` just return a `RefPtr<WebFrame>`? It seems like many cases want this.

brentfulgham reviewed (commented):
Looks good. Consider whether `WebFrame::fromCoreFrame` could always just return a RefPtr.

cdumez committed:
Use more smart pointers in Source/WebKit
https://bugs.webkit.org/show_bug.cgi?id=261474

Reviewed by Brent Fulgham.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::focusedElementChanged):
(WebKit::WebChromeClient::focusedFrameChanged):
(WebKit::WebChromeClient::createWindow):
(WebKit::WebChromeClient::runBeforeUnloadConfirmPanel):
(WebKit::WebChromeClient::closeWindow):
(WebKit::WebChromeClient::runJavaScriptAlert):
(WebKit::WebChromeClient::runJavaScriptConfirm):
(WebKit::WebChromeClient::runJavaScriptPrompt):
(WebKit::WebChromeClient::invalidateContentsAndRootView):
(WebKit::WebChromeClient::invalidateContentsForSlowScroll):
(WebKit::WebChromeClient::contentsSizeChanged const):
(WebKit::WebChromeClient::print):
(WebKit::WebChromeClient::runOpenPanel):
(WebKit::WebChromeClient::didAssociateFormControls):
(WebKit::WebChromeClient::showPlaybackTargetPicker):
(WebKit::WebChromeClient::hasStorageAccess):
(WebKit::WebChromeClient::requestStorageAccess):
(WebKit::WebChromeClient::shouldAllowDeviceOrientationAndMotionAccess):
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.cpp:
(WebKit::WebEditorClient::textFieldDidBeginEditing):
(WebKit::WebEditorClient::textFieldDidEndEditing):
(WebKit::WebEditorClient::textDidChangeInTextField):
(WebKit::WebEditorClient::textDidChangeInTextArea):
(WebKit::WebEditorClient::doTextFieldCommandFromEvent):
(WebKit::WebEditorClient::textWillBeDeletedInTextField):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction):
* Source/WebKit/WebProcess/WebCoreSupport/WebGeolocationClient.cpp:
(WebKit::WebGeolocationClient::geolocationDestroyed):
(WebKit::WebGeolocationClient::startUpdating):
(WebKit::WebGeolocationClient::stopUpdating):
(WebKit::WebGeolocationClient::setEnableHighAccuracy):
(WebKit::WebGeolocationClient::requestPermission):
(WebKit::WebGeolocationClient::revokeAuthorizationToken):
(WebKit::WebGeolocationClient::cancelPermissionRequest):
* Source/WebKit/WebProcess/WebCoreSupport/WebGeolocationClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didLoadFromRegistrableDomain):
(WebKit::WebLocalFrameLoaderClient::loadedSubresourceDomains const):
(WebKit::WebLocalFrameLoaderClient::detachedFromParent2):
(WebKit::WebLocalFrameLoaderClient::assignIdentifierToInitialRequest):
(WebKit::WebLocalFrameLoaderClient::dispatchWillSendRequest):
(WebKit::WebLocalFrameLoaderClient::shouldUseCredentialStorage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveResponse):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveContentLength):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishDataDetection):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishLoading):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFailLoading):
(WebKit::WebLocalFrameLoaderClient::dispatchDidDispatchOnloadEvents):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveServerRedirectForProvisionalLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidChangeProvisionalURL):
(WebKit::WebLocalFrameLoaderClient::dispatchDidCancelClientRedirect):
(WebKit::WebLocalFrameLoaderClient::dispatchWillPerformClientRedirect):
(WebKit::WebLocalFrameLoaderClient::dispatchDidChangeLocationWithinPage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidChangeMainDocument):
(WebKit::WebLocalFrameLoaderClient::dispatchWillChangeDocument):
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI):
(WebKit::WebLocalFrameLoaderClient::dispatchDidExplicitOpen):
(WebKit::WebLocalFrameLoaderClient::dispatchDidStartProvisionalLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReceiveTitle):
(WebKit::WebLocalFrameLoaderClient::dispatchDidCommitLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFailProvisionalLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFailLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishDocumentLoad):
(WebKit::WebLocalFrameLoaderClient::dispatchDidFinishLoad):
(WebKit::WebLocalFrameLoaderClient::completePageTransitionIfNeeded):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReachLayoutMilestone):
(WebKit::WebLocalFrameLoaderClient::dispatchDidLayout):
(WebKit::WebLocalFrameLoaderClient::dispatchCreatePage):
(WebKit::WebLocalFrameLoaderClient::dispatchShow):
(WebKit::WebLocalFrameLoaderClient::dispatchDecidePolicyForResponse):
(WebKit::WebLocalFrameLoaderClient::dispatchDecidePolicyForNewWindowAction):
(WebKit::WebLocalFrameLoaderClient::allowsContentJavaScriptFromMostRecentNavigation const):
(WebKit::WebLocalFrameLoaderClient::dispatchWillSendSubmitEvent):
(WebKit::WebLocalFrameLoaderClient::dispatchWillSubmitForm):
(WebKit::WebLocalFrameLoaderClient::willReplaceMultipartContent):
(WebKit::WebLocalFrameLoaderClient::didReplaceMultipartContent):
(WebKit::WebLocalFrameLoaderClient::finishedLoading):
(WebKit::WebLocalFrameLoaderClient::updateGlobalHistory):
(WebKit::WebLocalFrameLoaderClient::updateGlobalHistoryRedirectLinks):
(WebKit::WebLocalFrameLoaderClient::shouldGoToHistoryItem const):
(WebKit::WebLocalFrameLoaderClient::didDisplayInsecureContent):
(WebKit::WebLocalFrameLoaderClient::didRunInsecureContent):
(WebKit::WebLocalFrameLoaderClient::provisionalLoadStarted):
(WebKit::WebLocalFrameLoaderClient::setTitle):
(WebKit::WebLocalFrameLoaderClient::userAgent const):
(WebKit::WebLocalFrameLoaderClient::overrideContentSecurityPolicy const):
(WebKit::WebLocalFrameLoaderClient::transitionToCommittedForNewPage):
(WebKit::WebLocalFrameLoaderClient::createFrame):
(WebKit::WebLocalFrameLoaderClient::objectContentType):
(WebKit::WebLocalFrameLoaderClient::dispatchDidClearWindowObjectInWorld):
(WebKit::WebLocalFrameLoaderClient::dispatchGlobalObjectAvailable):
(WebKit::WebLocalFrameLoaderClient::dispatchServiceWorkerGlobalObjectAvailable):
(WebKit::WebLocalFrameLoaderClient::willInjectUserScript):
(WebKit::WebLocalFrameLoaderClient::dispatchWillDisconnectDOMWindowExtensionFromGlobalObject):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReconnectDOMWindowExtensionToGlobalObject):
(WebKit::WebLocalFrameLoaderClient::dispatchWillDestroyGlobalObjectForDOMWindowExtension):
(WebKit::WebLocalFrameLoaderClient::accessibilityRemoteObject):
(WebKit::WebLocalFrameLoaderClient::setAXIsolatedTreeRoot):
(WebKit::WebLocalFrameLoaderClient::willCacheResponse const):
(WebKit::WebLocalFrameLoaderClient::dataDetectionReferenceDate):
(WebKit::WebLocalFrameLoaderClient::didChangeScrollOffset):
(WebKit::WebLocalFrameLoaderClient::shouldForceUniversalAccessFromLocalURL):
(WebKit::WebLocalFrameLoaderClient::sendH2Ping):
(WebKit::WebLocalFrameLoaderClient::didRestoreScrollPosition):
(WebKit::WebLocalFrameLoaderClient::getLoadDecisionForIcons):
(WebKit::WebLocalFrameLoaderClient::didFinishServiceWorkerPageRegistration):
(WebKit::WebLocalFrameLoaderClient::notifyPageOfAppBoundBehavior):
(WebKit::WebLocalFrameLoaderClient::modelInlinePreviewUUIDs const):
* Source/WebKit/WebProcess/WebCoreSupport/WebValidationMessageClient.cpp:
(WebKit::WebValidationMessageClient::~WebValidationMessageClient):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebRemoteObjectRegistry.cpp:
(WebKit::WebRemoteObjectRegistry::WebRemoteObjectRegistry):
(WebKit::WebRemoteObjectRegistry::close):
(WebKit::WebRemoteObjectRegistry::messageSender):
(WebKit::WebRemoteObjectRegistry::messageDestinationID):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebRemoteObjectRegistry.h:
* Source/WebKit/WebProcess/WebPage/WebDocumentLoader.cpp:
(WebKit::WebDocumentLoader::loaderForWebsitePolicies):

Canonical link: https://commits.webkit.org/267945@main
